credentials/tls/certprovider/pemfile/watcher.go

```diff
@@ -55,22 +55,26 @@ var (
 type Options struct {
 	// CertFile is the file that holds the identity certificate.
 	// Optional. If this is set, KeyFile must also be set.
-	CertFile string
+	CertFile string `json:"certificate_file,omitempty"`
 	// KeyFile is the file that holds identity private key.
 	// Optional. If this is set, CertFile must also be set.
-	KeyFile string
+	KeyFile string `json:"private_key_file,omitempty"`
 	// RootFile is the file that holds trusted root certificate(s).
 	// Optional.
-	RootFile string
+	RootFile string `json:"ca_certificate_file,omitempty"`
 	// SPIFFEBundleMapFile is the file that holds the spiffe bundle map.
 	// If a given provider configures both the RootFile and the
 	// SPIFFEBundleMapFile, the SPIFFEBundleMapFile will be preferred.
 	// Optional.
-	SPIFFEBundleMapFile string
+	SPIFFEBundleMapFile string `json:"spiffe_trust_bundle_map_file,omitempty"`
 	// RefreshDuration is the amount of time the plugin waits before checking
 	// for updates in the specified files.
 	// Optional. If not set, a default value (1 hour) will be used.
 	RefreshDuration time.Duration
+	// RefreshInterval is the duration string parsed to set RefreshDuration.
+	// This field must be a valid Go duration string (e.g., "10s", "5m").
+	// If unset, the default value for RefreshDuration will be used.
+	RefreshInterval string `json:"refresh_interval,omitempty"`
```

Comment:
I don't think we want to introduce a new setting on this `Options` struct - this could be confusing for users - e.g. "Do I set RefreshDuration or RefreshInterval, and what's the difference?"

See
https://stackoverflow.com/questions/30445479/how-to-specify-default-values-when-parsing-json-in-go
and
https://stackoverflow.com/questions/48050945/how-to-unmarshal-json-into-durations

For handling default values when unmarshaling JSON

cc: @arjan-bal since he was the first to point this out

Reply:
@gtcooke94 I understand what you meaning, thx for the references.
so i wrap the RefreshDuration field as custom Duration type and removed confusing field(RefreshInterval).

but i'm bit worried that type of RefreshDuration would may change. how do you think of it?